src/helper.js

```diff
@@ -191,3 +193,43 @@ function mergeConfig(oldConfig, newConfig) {
 		}
 	});
 }
+
+function parseHostmask(hostmask) {
+	let nick = "";
+	let ident = "*";
+	let hostname = "*";
+	let parts = [];
+
+	// Parse hostname first, then parse the rest
+	parts = hostmask.split("@");
+
+	if (parts.length >= 2) {
+		hostname = parts[1] || "*";
+		hostmask = parts[0];
+	}
+
+	parts = hostmask.split("!");
+
+	if (parts.length >= 2) {
+		ident = parts[1] || "*";
+		hostmask = parts[0];
+	}
+
+	nick = hostmask || "*";
+
+	const result = {
+		nick: nick,
+		ident: ident,
+		hostname: hostname,
+	};
+
+	result.toString = function() {
```

Comment:
This should be removed, as it more than likely will break after being JSON encoded (e.g. server restart). Just format it where it's needed.